go/def.bzl

```diff
@@ -217,19 +217,39 @@ def emit_go_compile_action(ctx, sources, deps, out_lib, extra_objects=[]):
   if _is_external(out_dir):
     out_depth -= 2
   cmds = symlink_tree_commands(out_dir, tree_layout)
+
+  extra_inputs = ctx.files.toolchain
+
+  # cd into the out_dir.
+  cmds.append("export GOROOT=$(pwd)/" + ctx.file.go_tool.dirname + "/..")
+  cmds.append(" ".join(["cd", out_dir]))
+
+  # Apply build tags here.
+  use_build_tags = hasattr(ctx.attr, "filter_tags")
```

Comment:
If _filter_tags is in go_env_attrs (see comment below), then you won't need this test. Just assume it's there.